presidio-analyzer/analyzer/pattern_recognizer.py

```diff
@@ -120,7 +121,7 @@ def __analyze_patterns(self, text):
                                        pattern.strength)
                 res = self.validate_result(current_match, res)
 
-                if res:
+                if res and res.score != 0:
```

Comment:
That's what I fixed. we are not returning results if the score = 0

Reply:
The reason we didn't filter result with 0 socre, is for the future support in the context.
When it will be supported, there are some results that without the context don't mean anything, but with the context can be meanningful.
(For example driver license: 123456)  

If I rembmber currectly, the AnalyzerEngine filters the results with socre 0

Reply:
@Ilana - I prefer to filter 0 score results in the recognizer level. We should address the context in each recognizer, similarly to the way we do validation. 
Bottom line - our recognizers should be context aware and provide the final score per each entity they support.

Reply:
@limorl even if recognizers are context aware, that would happen in a later phase in the analysis process. In this case it would be removed and context wouldn't matter. I think a good way to tackle this is not to have regex patterns with 0 score. Even if this is a very weak patter, it should have a score of 0.01 or 0.05. Then context would be able to extend this. We can further have "regex strength" and "context strength" parameters for pattern recognizers designers to decide upon.

Reply:
@limorl @Ilanak The current design doesn't assume recognizers are context aware. We'll have a discussion around changing. Current design assumes that the context extraction is an independent process happening globally across all recognizers.

Reply:
Discussed offline, the it is valid to filter out 0-score results. In a later PR we are re-enabling context.